kdd-2024/ibis-tutorial-KDD2024.md

```diff
@@ -0,0 +1,273 @@
+
+**Expertise:** 
+Gil Forsyth is an experienced instructor, having led tutorials at several PyData conferences, PyCon, and SciPy. He also ran internal training on Python and distributed data analysis at Capital One for several years. Gil is one of the core maintainers of Ibis.
+
+The most recent Ibis tutorial is the accepted Ibis tutorial at PyCon US 2024: 
+[Tutorials: Introduction to Ibis: blazing fast analytics with DuckDB, Polars, Snowflake, and more, from the comfort of your Python repl.](https://us.pycon.org/2024/schedule/presentation/55/)
+
+
+2. List of contributors, i.e., tutors who will only help prepare the tutorial material
+
+- I'm not sure who to put here, I will need designated responsible people for some 
+  Ibis ML showcase and Ibis streaming (Chloe?)
+
+
+# Corresponding tutor with her/his email address
+
+- Naty Clementi (I'll add my email upon submission)
+
+# Tutorial outline. Please provide as much detail as possible.
+
+Here is where I need some help. I modify parts of the Scipy outline to include more about 
+why Ibis, what is Ibis and for Who, I think this will let us introduce Ibis as part 
+of a "composable data ecosystem" (also requested by Ian).  We need to include 
+some streaming content and some ML content, again keep in mind that the ask is 
+to introduce this as a 20% of total content.
+
+## 0:00 - Intro and Setup “Going beyond pandas”
+
+Get attendees up and running in a GitHub Codespace or on their laptops. A bit of
+motivation about the kinds of problems where Ibis can help, and a general survey 
+of attendees to find out what their existing pain points and experiences are.
+
+## 0:15 - Introduction to Ibis basics
+
+**Slides**
+- What is Ibis?: Ibis as part of a composable data ecosystem.
+- Why Ibis?: present the problems that Ibis tackles and the benefits of adopting it.
+- Who Is Ibis for?: shown the versatility of profiles that can benefit from Ibis (data engineers, data analysts, and data scientists, any data practitioner) and how.
+
+**Get familiarized with basics: **
+
+A hands-on, follow-along notebook introducing the basic verbs of Ibis data 
+analysis (select, filter, group_by, order_by, and aggregate), demonstrating a 
+simple example on a few local query engines with hands-on exercises throughout.
+
+## 1:00 - Coffee Break (5 minutes that definitely takes 10 minutes)
+
+## 1:10 - In-memory tables, joins, and data analysis (need to adapt this for audience)
+
+Building on the previous notebook, we'll explore how to join in-memory data (from 
+a pandas DataFrame, Python dictionary, or PyArrow Table) with existing tables in 
+a local database and continue analysis on the join result.
+
+We'll touch on the Ibis deferred operator for specifying predicates in chained joins, 
+and demonstrate read_parquet and other read_* methods for loading local data into
+existing databases. Then we'll continue with a series of hands-on exercises, 
```

Comment:
```suggestion
We'll explore chained joins, and demonstrate read_parquet and other read_* methods for loading local data into
existing databases. Then we'll continue with a series of hands-on exercises, 
```

we no longer _need_ the underscore for chained joins, so we can introduce it elsewhere.